explainaboard/utils/span_utils.py

```diff
@@ -156,318 +225,113 @@ def get_matched_spans(
-        self,
-        resources: dict[str, Any] = {},
-        match_type: str = "tag",
-        n_grams: list = [1, 2],
-    ):
-        super().__init__(resources, match_type)
-        self.n_grams = n_grams
-
-    @classmethod
-    def default_match_type(cls) -> str:
-        return "tag"
-
-    def get_spans_and_match(self, tags: list, tags_other: list):
-        def get_ngrams(tags, n_grams: list[int]):
-            spans = []
-            for k in n_grams:
-                for i, tok in enumerate(tags):
-                    if i + k > len(tags):
-                        break
-                    span = " ".join(tags[i : i + k])
-                    spans.append((span, i, i + k))
-            return spans
-
-        def get_span_from_ngrams(ngrams, tags_other_table, tags_length):
-            span_dics = []
-            for ngram in ngrams:
-                span = ngram[0]
-
-                # match
-                my_other = tags_other_table.get(span, list())
-                matched = my_other.pop(0) if len(my_other) > 0 else -1
-
-                span_dic = Span(
-                    span_text=span,
-                    span_tag=span
-                    if "span_tag" not in self.resources.keys()
-                    else self.resources["span_tag"](span),
-                    span_pos=(ngram[1], ngram[2]),
-                    span_matched=matched,
-                    span_capitalness=cap_feature(span),  # type: ignore
-                    span_rel_pos=ngram[2] * 1.0 / tags_length,  # type: ignore
-                    span_chars=len(span),
-                    span_tokens=len(span.split(" ")),
-                    span_test_freq=0
-                    if "ref_test_freq" not in self.resources.keys()
-                    else self.resources["ref_test_freq"].get(span, 0),  # type: ignore
-                    span_train_freq=0
-                    if "fre_dic" not in self.resources.keys()
-                    or self.resources["fre_dic"] is None
-                    else self.resources["fre_dic"].get(span, 0),  # type: ignore
-                )
-                # Save the features
-                span_dics.append(span_dic)
-            return span_dics
-
-        tags_ngrams = get_ngrams(tags, self.n_grams)
-        tags_other_ngrams = get_ngrams(tags_other, self.n_grams)
-
-        # Find tokens in other set
-        tags_other_table = defaultdict(list)
-        for i, tok in enumerate(tags_other_ngrams):
-            tags_other_table[tok[0]].append(i)
-
-        # Find tokens in other set
-        tags_table = defaultdict(list)
-        for i, tok in enumerate(tags_ngrams):
-            tags_table[tok[0]].append(i)
-
-        span_dics = get_span_from_ngrams(tags_ngrams, tags_other_table, len(tags))
-        span_dics_other = get_span_from_ngrams(
-            tags_other_ngrams, tags_table, len(tags_other)
-        )
-
-        return span_dics, span_dics_other
+# TODO(gneubig): this is not used anywhere, so decide to keep or delete
```

Comment:
Aha, this is originally expected to be applied to conditional text generation tasks, extending token-level accuracy to span-level

Reply:
I'll leave it commented out for now, but we can either delete or resurrect it later.